Avoid constructing an empty tuple in qsearch (obsolete)

snicolet opened the pull request:
Avoid constructing, passing as a parameter and binding a useless empty tuple of pointers in the qsearch move picker constructor. Also add some comments in movepicker.cpp and do some cleaning in evaluate.cpp while there.

No functional change.

joergoster commented:
Looks good. But why dropping the comment for TrappedBishopA1H1 penalty?

snicolet commented:
@joergoster The comment for TrappedBishopA1H1 in lines 220-222 is redundant with the comment in lines 348-350.

joergoster commented:
@snicolet You're right. Sorry for the noise.

mcostalba commented:
@snicolet fine for me. Can you please squash in  single commit? Thanks.

While there, if you don't mind, can you please update also [MovePicker::score() ](https://github.com/mcostalba/Stockfish/blob/62fe0449cacf7f4f2a64e51310488e93b4e67917/src/movepick.cpp#L127)?

See related discussion [here](https://github.com/official-stockfish/Stockfish/pull/1199#issuecomment-323569633)

snicolet commented:
@mcostalba 

I have updated the scoring function, squashed all the changes into a single commit and created another pull request there: https://github.com/official-stockfish/Stockfish/pull/1220

So I'm closing this one as it is now obsolete.
